request.js

```diff
@@ -113,9 +130,13 @@
             if(method === 'GET'){
                 data = null;
             }
-            setTimeout(function(){
```

Comment:
That's the magic of the `defer`. The `setTimeout` is because there was not `defer`, so we had to give the user the possibility to set the callbacks before doing the request; with Q, even if the request is resolved before returning the promise, the way Q works, once you define the `.then` it will be called (Q has an internet interval, you've probably seen it as `nextTick` on the debugger).